src/sentry/spans/consumers/process/factory.py

```diff
@@ -0,0 +1,55 @@
+import logging
+from collections.abc import Mapping
+from typing import Any
+
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.commit import CommitOffsets
+from arroyo.processing.strategies.run_task import RunTask
+from arroyo.types import BrokerValue, Commit, Message, Partition
+from sentry_kafka_schemas import get_codec
+from sentry_kafka_schemas.codecs import Codec
+from sentry_kafka_schemas.schema_types.snuba_spans_v1 import SpanEvent
+
+from sentry.spans.buffer.redis import RedisSpansBuffer
+from sentry.tasks.spans import process_segment
+
+logger = logging.getLogger(__name__)
+SPAN_SCHEMA: Codec[SpanEvent] = get_codec("snuba-spans")
+
+PROCESS_SEGMENT_DELAY = 2 * 60  # 2 minutes
+
+
+def _deserialize_span(value: bytes) -> Mapping[str, Any]:
+    return SPAN_SCHEMA.decode(value)
+
+
+def process_message(message: Message[KafkaPayload]):
+    assert isinstance(message.value, BrokerValue)
+    try:
+        span = _deserialize_span(message.payload.value)
+        segment_id = span["segment_id"]
+        project_id = span["project_id"]
+    except Exception:
+        logger.exception("Failed to process span payload")
+
+    client = RedisSpansBuffer()
+    new_segment = client.write_span(project_id, segment_id, message.payload.value)
```

Comment:
You may want to be careful here. This instantiates a new redis client per message/span and does a single write to redis. Depending on your consumer configurations, you may end up with a lot of connections and you'd be opening/closing them constantly.

Reply:
Actually, maybe we already reuse the clients in `sentry.utils.redis` 🤔 but if it's possible to bulk write to Redis, we should consider that.

Reply:
yeah, will consider that for sure! starting with basic scaffolding here